src/paho/mqtt/client.py

```diff
@@ -2124,6 +2148,8 @@ def _handle_publish(self):
         elif message.qos == 2:
             rc = self._send_pubrec(message.mid)
             message.state = mqtt_ms_wait_for_pubrel
+            if self._max_in_messages > 0 and len(self._in_messages) >= self._max_in_messages:
```

Comment:
I think returning an error here will disconnect the client, which seems like a bad outcome.